### PR TITLE
Set user.room for reaction_added events

### PR DESCRIPTION
**Pull request by mbland**
This addresses 18F/hubot-slack-github-issues#33.

cc: @afeld

### CODE REVIEW

File: src/slack.coffee [hunk 1]
```diff
@@ -174,7 +174,10 @@ class SlackBot extends Adapter
       @receive new SlackTextMessage user, text, rawText, msg
 
   reactionAdded: (msg) =>
+    return if msg.user == @self.id
```
Comment (afeld) [inline src/slack.coffee:178]: I don't really understand what this line is doing...maybe worth an inline comment?

**Comment by afeld** [comment]
Were these modified tests failing without the associated code change? Seems like they may not actually be testing the behavior described in the linked issue.

File: src/slack.coffee (hunk 1 above)
Comment (mbland) [inline src/slack.coffee:178]: Done.

**Comment by mbland** [comment]
The updated `should produce a SlackTextMessage` and `should parse the text in the SlackTextMessage` tests pass without any changes to `SlackBot.message` because it was already setting `user.room`. The `should produce a SlackTextMessage for reaction_added events` test failed before updating `SlackBot.reactionAdded` because `user.room` was previously undefined.

My hypothesis is that setting `user.room` in `SlackBot.reactionAdded` addresses 18F/hubot-slack-github-issues#33 because it's the only context I've discovered that could account for a Hubot reply going to the channel that a user last commented in, rather than to the channel in which the `reaction_added` event took place. In fact...

I just deployed a Charlie with this change, and 18F/handbook#443 confirms that it works.

**Pull request merged**